core/frontend/src/types/system-information/system.ts

```diff
@@ -0,0 +1,193 @@
+/** Base structure that provides CPU specific information */
+export interface CPU {
+    /**
+     * @param brand - E.g: "ARMv7 Processor rev 3 (v7l)"
+     * @param frequency - Frequency in MHz
+     * @param name - E.g: "cpu0"
+     * @param usage - Usage between 0-100
+     * @param vendor_id - E.g: "AuthenticAMD"
+    * */
+    brand: string,
+    frequency: number,
+    name: string,
+    usage: number,
+    vendor_id: string
+}
+
+/** Base structure that provides disk specific information */
+export interface Disk {
+    /**
+     * @param available_space_B - Free space in bytes
+     * @param filesystem_type - E.g: Ext4, NTFS
+     * @param mount_point - E.g: /mnt/HD, /
+     * @param name - Name that is used to identify the disk, E.g: /dev/root, /dev/sda1
+     * @param total_space_B - Size of the disk in bytes
+     * @param type - E.g: SSD, HD
+    * */
+    available_space_B: number,
+    filesystem_type: string,
+    mount_point: string,
+    name: string,
+    total_space_B: number,
+    type: string
+}
+
+/** Base structure that provides basic OS/computer information */
+export interface Info {
```

Comment:
Will this conflict with the other Info interface?

Reply:
Only if you include it, but you can also include with a different name.
`import { info as SystemInfo } ...`